src/spikeinterface/core/zarrextractors.py

```diff
@@ -0,0 +1,424 @@
+            zarr_path=zarr_path,
+            storage_options=storage_options,
+            dataset_paths=dataset_paths,
+            **zarr_kwargs,
+            **job_kwargs,
+        )
+
+        # # save probe
+        if recording.get_property("contact_vector") is not None:
+            probegroup = recording.get_probegroup()
+            zarr_root.attrs["probe"] = check_json(probegroup.to_dict(array_as_list=True))
+
+        # save time vector if any
+        t_starts = np.zeros(recording.get_num_segments(), dtype="float64") * np.nan
+        for segment_index, rs in enumerate(recording._recording_segments):
+            d = rs.get_times_kwargs()
+            time_vector = d["time_vector"]
+            if time_vector is not None:
+                _ = zarr_root.create_dataset(
+                    name=f"times_seg{segment_index}",
+                    data=time_vector,
+                    filters=zarr_kwargs.get("filters", None),
+                    compressor=zarr_kwargs["compressor"],
+                )
+            elif d["t_start"] is not None:
+                t_starts[segment_index] = d["t_start"]
+
+        if np.any(~np.isnan(t_starts)):
+            zarr_root.create_dataset(name="t_starts", data=t_starts, compressor=None)
+
+        add_properties_and_annotations(zarr_root, recording)
+
+
+class ZarrRecordingSegment(BaseRecordingSegment):
+    def __init__(self, root, dataset_name, **time_kwargs):
+        BaseRecordingSegment.__init__(self, **time_kwargs)
+        self._timeseries = root[dataset_name]
+
+    def get_num_samples(self) -> int:
+        """Returns the number of samples in this signal block
+
+        Returns:
+            SampleIndex: Number of samples in the signal block
+        """
+        return self._timeseries.shape[0]
+
+    def get_traces(
+        self,
+        start_frame: Union[int, None] = None,
+        end_frame: Union[int, None] = None,
+        channel_indices: Union[List, None] = None,
+    ) -> np.ndarray:
+        traces = self._timeseries[start_frame:end_frame]
+        if channel_indices is not None:
+            traces = traces[:, channel_indices]
+        return traces
+
+
+class ZarrSortingExtractor(BaseSorting):
+    """
+    SortingExtractor for a zarr format
+
+    Parameters
+    ----------
+    root_path: str or Path
+        Path to the zarr root file
+    storage_options: dict or None
+        Storage options for zarr `store`. E.g., if "s3://" or "gcs://" they can provide authentication methods, etc.
+
+    Returns
+    -------
+    sorting: ZarrSortingExtractor
+        The sorting Extractor
+    """
+
+    extractor_name = "ZarrSorting"
+    installed = HAVE_ZARR  # check at class level if installed or not
+    mode = "file"
+    # error message when not installed
+    installation_mesg = "To use the ZarrSortingExtractor install zarr: \n\n pip install zarr\n\n"
+    name = "zarr"
+
+    def __init__(self, root_path: Union[Path, str], storage_options: dict | None = None):
```

Comment:
What do you think of unify and just call everything a `file_path` or `folder_path`. That way, if we want to use keyword arguments we don't need to remember specific keywords for every different format. Does that makes sense here?

Reply:
Yep, nothing against it! Refactoring :)

Reply:
done